### DISCUSSION TITLE
Pass It On alternative

### DISCUSSION THREAD

**Discussion by aripuguh** [GECO-Project-App/facilitation-app#71] [3 - Design]
Hej hej! **Brainstorming Pass It On** alternative here. @eliasfaltin 

What do you think instead of having 3 different pages, the steps can be just on one page with a scrollable bar where the user can return to the previous/go to the next step without really having the hassle of needing to remember what they had to do and click _the next step_. I think after trying out the app, I felt like I could easily forget what I had to do in the previous step (particularly on the pass-it method). The objective of this new proposal is so the user can easily view/recap all the needed steps, all in one page. [Here is a prototype on the suggestion](https://www.figma.com/proto/8ZmfzVPn0pKOKNyRITviyk/Facilitation-App?page-id=479%3A980&node-id=479-981&node-type=canvas&viewport=334%2C382%2C0.69&t=55BQHnVqTxq2N7WZ-1&scaling=min-zoom&content-scaling=fixed) (don’t worry so much on the graphic, I’ll make more interesting rive animation). LMK if it made sense. Feel free to share your frank feedback!
<img width="861" alt="Screen Shot 2024-09-20 at 6 49 36 PM" src="https://github.com/user-attachments/assets/1475e7f4-1219-4e7b-8a76-328ccb9bc9ce">

**Comment by alice-ang** [comment]
Yeah I like this. I had a similar feeling something needed to be changed with the pass it on method. I was thinking we could probably utilize something like the carousel component in shadcn https://ui.shadcn.com/docs/components/carousel (and heavily customise if ofc) to achieve something like this!

**Reply by eliasfaltin** [reply]
Or we do the whole thing in Rive and just use a "step" int variable that we pass to the animation

**Reply by aripuguh** [reply]
<img width="826" alt="Screen Shot 2024-09-20 at 8 32 09 PM" src="https://github.com/user-attachments/assets/fc84ef4a-1427-4a9f-b99b-710b34dafdfc">

[Revised version](https://www.figma.com/proto/8ZmfzVPn0pKOKNyRITviyk/Facilitation-App?page-id=479%3A980&node-id=479-981&node-type=canvas&viewport=334%2C382%2C0.69&t=omBuk60zkDRqpn3Z-1&scaling=min-zoom&content-scaling=fixed) based on your feedback on Slack. LMK if I missed anything/have some thoughts.

**Reply by aripuguh** [reply]
<img width="375" alt="Screen Shot 2024-09-20 at 10 50 02 PM" src="https://github.com/user-attachments/assets/060f528d-9d57-423c-86e2-1be38bdc5501">
Or do you prefer this arrow style?

**Reply by aripuguh** [reply]
Or this 
<img width="392" alt="Screen Shot 2024-09-21 at 8 59 51 AM" src="https://github.com/user-attachments/assets/57ffc3c8-6934-4ee1-b4eb-d3cc3f99c98c">

**Reply by eliasfaltin** [reply]
I like the arrows/navigation below the animation. The slider concept wasn't super intuitive. It looked more like an indicator/loading bar than something a user would click.

We could have arrows below the animation and the stepper as a part of the exercise header component, as in other exercises. Or would that defeat the purpose of "containing" the animations on "one page"?



Regarding which arrow option:
- I think there is an argument to re-using the same arrow we are using for navigation so that we don't clutter the design with 15 different arrow types over time
- there is another argument for using a distinct different arrow style since it is not for page navigation in this case.

**Reply by aripuguh** [reply]
Thanks for the feedback! 
Do you mean like this? 
<img width="889" alt="Screen Shot 2024-09-23 at 5 32 51 PM" src="https://github.com/user-attachments/assets/c0708b9d-babe-4394-b20a-b2cd2d8018c3">

LMK your thought.

**Comment by eliasfaltin** [comment]
<img width="534" alt="{2DC81A12-D10B-489B-98B1-82DBB3C4A6C5}" src="https://github.com/user-attachments/assets/27207741-6cdf-4bdb-921e-432c1348d010">
Final design
